### PR TITLE
[Preview] Recognize React.Component as a parent class.

### PR DESCRIPTION
**Pull request by adancarrasco**
Fixes Issue: #6331 

### Summary of Changes

* Pretty much followed @jasonLaster comments on the issue. 

### Test Plan
- Open https://firefox-debugger-example-react-js.glitch.me/ debugger test page
- Select parse-worker from workers
- Command-P opens the panel and look for getSymbols.js
- Put a breakpoint in line 148
- Reload the target debug page
- It'll hit the breakpoint, then step over
- `symbols.classes` array has the `UnorderedList` object with the parent as shown in the screenshot

### Screenshots/Videos
Object in parser-worker.js, parent added as React.Component
image

### CODE REVIEW

File: src/actions/tests/__snapshots__/preview.spec.js.snap [hunk 1]
```diff
@@ -46,14 +46,7 @@ exports[`setPreview react instance 1`] = `
 Object {
   "cursorPos": undefined,
   "expression": "this",
-  "extra": Object {
```
Comment (adancarrasco) [inline src/actions/tests/__snapshots__/preview.spec.js.snap]: After I updated `inComponent.js` this got empty. I don't think this should be the right snapshot. I'll investigate later.

Comment (darkwing) [reply src/actions/tests/__snapshots__/preview.spec.js.snap]: Yep -- that's an issue.

File: src/workers/parser/getSymbols.js [hunk 2]
```diff
@@ -33,7 +34,7 @@ export type SymbolDeclaration = {
 };
 
 export type ClassDeclaration = SymbolDeclaration & {
-  parent: string
+  parent: Object | null
```
Comment (jasonLaster) [inline src/workers/parser/getSymbols.js]: we should improve the flow type here:

```
name: string,
location: Location
```

Comment (adancarrasco) [reply src/workers/parser/getSymbols.js]: Thanks for the feedback Jason, I would work on it this week. :)

File: src/actions/tests/__snapshots__/preview.spec.js.snap (hunk 1 above)
Comment (adancarrasco) [reply src/actions/tests/__snapshots__/preview.spec.js.snap]: Thanks @darkwing. I will review it this week. :)

**Comment by darkwing** [comment]
How's it going @adancarrasco ?  Anything we can help with?

**Comment by jasonLaster** [comment]
Thanks @adancarrasco!

**Comment by adancarrasco** [comment]
Thank you guys, sorry for the delay. I was off and couldn't make it as promised.

**Pull request merged**